synapseclient/core/utils.py

```diff
@@ -173,8 +173,9 @@ def extract_filename(content_disposition_header, default_filename=None):
 
     if not content_disposition_header:
         return default_filename
-    value, params = cgi.parse_header(content_disposition_header)
-    return params.get("filename", default_filename)
+    message = Message()
+    message.add_header("content-disposition", content_disposition_header)
+    return message.get_filename(failobj=default_filename)
```

Comment:
I verified that there was unit test coverage that already covered this function.